sys/ztimer/periph_rtt.c

```diff
@@ -74,17 +74,42 @@ static void _ztimer_periph_rtt_cancel(ztimer_clock_t *clock)
     rtt_clear_alarm();
 }
 
+#if MODULE_ZTIMER_ONDEMAND_RTT
+static void _ztimer_periph_rtt_start(ztimer_clock_t *clock)
+{
+    (void)clock;
+    rtt_poweron();
+}
+
+static void _ztimer_periph_rtt_stop(ztimer_clock_t *clock)
+{
+    (void)clock;
+    rtt_poweroff();
+}
+#endif /* MODULE_ZTIMER_ONDEMAND_RTT */
```

Comment:
```suggestion
__attribute__((unused))
static void _ztimer_periph_rtt_start(ztimer_clock_t *clock)
{
    (void)clock;
    rtt_poweron();
}

__attribute__((unused))
static void _ztimer_periph_rtt_stop(ztimer_clock_t *clock)
{
    (void)clock;
    rtt_poweroff();
}
```

Would be an alternative. The advantage would be that the start and stop functions would be compile-tested no matter what. The `__attribute__((unused))` is pretty ugly, though.

If you take the suggestion, please do so consistently for all clock backends.

Reply:
https://github.com/RIOT-OS/RIOT/pull/18884 👀

Reply:
I've thought about this. It'll add compile time for systems that doesn't make use of `ztimer_ondemand_rtt`. Do we want that cost? I'm not sure ...

Reply:
Are you okay to address this in a follow-up? There's a lot of code in `ztimer` applying the same pattern.

Reply:
Yes! Let's get this in first and address the nit picks afterwards :) I may just open a PR to address this here and elsewhere in ztimer :)